### ISSUE TITLE
请问mode=='evaluate'时，run_evaluate函数里计算的metric只是一个batch的metric？

### ISSUE THREAD

**Issue by Ljl-Jdsk** [alibaba/euler#195]
请问mode=='evaluate'时，run_evaluate函数里计算的metric只是一个batch的metric 吗？
如下代码，
    with tf.train.MonitoredTrainingSession(
            master=master, is_chief=is_chief,
            checkpoint_dir=flags_obj.model_dir,
            save_checkpoint_secs=None,
            log_step_count_steps=None,
            hooks=hooks, config=config) as sess:
        while not sess.should_stop():
            metric_val = sess.run(metric)

    print('{}: {}'.format(metric_name, metric_val))
里面没有加hook，所以source没有next时自动停止吗？
每一次batch，返回的metric都是一个具体值，这样最后metric_val 就是最后一个batch计算出的
metric，请问是这样的吗？
